Shadowlands/TazaveshTheVeiledMarket/Hylbrande.lua

```diff
@@ -48,33 +59,69 @@ function mod:UNIT_SPELLCAST_SUCCEEDED(_, _, _, spellId)
 	if spellId == 346971 then -- [DNT] Summon Vault Defender
 		self:Message(spellId, "yellow", CL.add_spawned)
 		self:PlaySound(spellId, "info")
-		if self:BarTimeLeft(346766) > 29.1 then -- Sanitizing Cycle
+		if sanitizingCycleTime - GetTime() > 29.1 then -- Sanitizing Cycle
 			self:Bar(spellId, 29.1, CL.adds)
 		end
 	end
 end
 
+function mod:PurifyingBurst(args)
+	self:Message(args.spellId, "yellow")
+	self:PlaySound(args.spellId, "info")
+	if sanitizingCycleTime - GetTime() > 21.8 then -- Sanitizing Cycle
+		self:CDBar(args.spellId, 21.8)
+	end
+end
+
 function mod:ShearingSwings(args)
 	self:Message(args.spellId, "purple")
 	self:PlaySound(args.spellId, "alarm")
-	if self:BarTimeLeft(346766) > 10.9 then -- Sanitizing Cycle
+	if sanitizingCycleTime - GetTime() > 10.9 then -- Sanitizing Cycle
 		self:CDBar(args.spellId, 10.9)
 	end
 end
 
 function mod:TitanicCrash(args)
 	self:Message(args.spellId, "red")
 	self:PlaySound(args.spellId, "alarm")
-	if self:BarTimeLeft(346766) > 23.1 then -- Sanitizing Cycle
+	if sanitizingCycleTime - GetTime() > 23.1 then -- Sanitizing Cycle
 		self:Bar(args.spellId, 23.1)
 	end
 end
 
-function mod:PurgedByFire(args)
-	self:Message(args.spellId, "orange")
-	self:PlaySound(args.spellId, "alert")
-	if self:BarTimeLeft(346766) > 17 then -- Sanitizing Cycle
-		self:Bar(args.spellId, 17)
+do
+	local prev = 0
+	function mod:CHAT_MSG_RAID_BOSS_WHISPER(_, _, _, _, _, playerName)
```

Comment:
Does this need to be scoped further? GUID is nil on this event unfortunately. I can get the boss name (Titanic Defense Turret) but there's localization issues scoping this. Any way to convert from localized name to NPC ID?

Reply:
No, we try to avoid using such events and get Blizz to add actual debuffs to players where possible. Sometimes it's unavoidable, so it comes down to using various tricks such as "is this the only whisper in the entire encounter" then good we don't need to bother checking localization of a boss name or message.

It's flawed because, for example, it could target a pet with the same name as a player, but we just have to deal with it.

Reply:
Thanks for the info!

This one is weird because it was added in a hotfix and only gives the whisper to the player who is targeted, so there's no way (for LittleWigs) to tell who its on if it's not on you.

But since this is the only whisper in the encounter and it gets unregistered after I think it's all good there.

Reply:
I should add that using the event is unavoidable in this case, there's no debuff or anything on the targeted player.